### DISCUSSION TITLE
Scenario stubs generated with underscores _

### DISCUSSION THREAD

**Discussion by Vic152** [pytest-dev/pytest-bdd#657] [Q&A]
```python
@then(parsers.parse("options {options} are available"))
def _(options):
    """options <options> are available."""
```

Is it normal that pytest-bdd code generator does not generate method names? I am coming back after few years and remember it generating method names to more less match the step.

**Comment by youtux** [comment]
yes, function names for steps implementation are meaningless, since all the info is already represented in the argument for `@then(...)` decorator, and you are not going to call these functions anyway. I took inspiration from the [ward test framework](https://ward.readthedocs.io).

https://github.com/pytest-dev/pytest-bdd/pull/536
